src/main/java/com/pngencoder/PngEncoder.java

```diff
@@ -11,14 +11,30 @@
 import java.nio.file.Paths;
 import java.util.zip.Deflater;
 
+/**
+ * Main class, containing the interface for PngEncoder.
+ * PngEncoder is a really fast encoder for PNG images in Java.
+ */
 public class PngEncoder {
+    /**
+     * Compression level 9 is the default.
+     * It produces images with a size comparable to ImageIO.
+     */
     public static int DEFAULT_COMPRESSION_LEVEL = Deflater.BEST_COMPRESSION;
 
     private final BufferedImage bufferedImage;
     private final int compressionLevel;
     private final boolean multiThreadedCompressionEnabled;
     private final PngEncoderSrgbRenderingIntent srgbRenderingIntent;
 
+    /**
```

Comment:
This constructor is private. Documenting it seems overkill.

Reply:
Oh, I meant to document public only. I guess I got carried away. :-)